pkg/kv/kvserver/replica_lifecycle_datadriven_test.go

```diff
@@ -257,50 +244,118 @@ func TestReplicaLifecycleDataDriven(t *testing.T) {
 				rs.lease = lease
 				return "ok"
 
-			case "run-split-trigger":
+			case "eval-split":
 				rangeID := dd.ScanArg[roachpb.RangeID](t, d, "range-id")
-				split, ok := tc.splits[rangeID]
-				require.True(t, ok, "split trigger not found for range-id %d", rangeID)
+				splitKey := roachpb.RKey(dd.ScanArg[string](t, d, "split-key"))
+				verbose := d.HasArg("verbose")
 				rs := tc.mustGetRangeState(t, rangeID)
 				desc := rs.desc
+				require.True(
+					t,
+					desc.RSpan().ContainsKey(splitKey),
+					"split key not within range",
+				)
+				leftDesc := desc
+				leftDesc.EndKey = splitKey
+				rightDesc := desc
+				rightDesc.RangeID = tc.nextRangeID
+				tc.nextRangeID++
+				rightDesc.StartKey = splitKey
+				split := roachpb.SplitTrigger{
+					LeftDesc:  leftDesc,
+					RightDesc: rightDesc,
+				}
 
-				return tc.mutate(t, func(batch storage.Batch) {
-					rec := (&batcheval.MockEvalCtx{
-						ClusterSettings:        tc.st,
-						Desc:                   &desc,
-						Clock:                  tc.clock,
-						AbortSpan:              rs.abortspan,
-						LastReplicaGCTimestamp: rs.lastGCTimestamp,
-						RangeLeaseDuration:     tc.rangeLeaseDuration,
-					}).EvalContext()
-
-					in := batcheval.SplitTriggerHelperInput{
-						LeftLease:      rs.lease,
-						GCThreshold:    &rs.gcThreshold,
-						GCHint:         &rs.gcHint,
-						ReplicaVersion: rs.version,
-					}
-					_, _, err := batcheval.TestingSplitTrigger(
-						ctx, rec, batch, enginepb.MVCCStats{}, split, in, hlc.Timestamp{},
-					)
+				// Run the split trigger evaluation and capture the batch that's
+				// generated for replication. Stash it away. This represents the
+				// raft log entry that will be applied as part of split
+				// application.
+				batch := tc.storage.NewBatch()
+				rec := (&batcheval.MockEvalCtx{
+					ClusterSettings:        tc.st,
+					Desc:                   &desc,
+					Clock:                  tc.clock,
+					AbortSpan:              rs.abortspan,
+					LastReplicaGCTimestamp: rs.lastGCTimestamp,
+					RangeLeaseDuration:     tc.rangeLeaseDuration,
+				}).EvalContext()
+
+				in := batcheval.SplitTriggerHelperInput{
+					LeftLease:      rs.lease,
+					GCThreshold:    &rs.gcThreshold,
+					GCHint:         &rs.gcHint,
+					ReplicaVersion: rs.version,
+				}
+				_, _, err := batcheval.TestingSplitTrigger(
+					ctx, rec, batch, enginepb.MVCCStats{}, &split, in, hlc.Timestamp{},
+				)
+				require.NoError(t, err)
+
+				batchRepr := batch.Repr()
+				tc.splits[rangeID] = pendingSplit{
+					trigger:   split,
+					batchRepr: batchRepr,
+				}
+				tc.updatePostSplitRangeState(ctx, t, batch, split)
```

Comment:
In relation to the earlier comment about updating the LHS/RHS in-memory state: in reality, this happens later as part of applying the split (it's kind of a transaction that updates both storage and in-memory map of replicas). Ideally, we should strive to mimic the `Store` behaviour.

Reply:
See my comment re: meta1/2.